packages/api/src/providers/mapper/mapper.provider.ts

```diff
@@ -0,0 +1,38 @@
+import {Mapper, MapperConstructor} from './mapper';
+
+/**
+ * A utility class that provides and manages singleton instances of `Mapper` objects.
+ *
+ * The `MapperProvider` is designed to handle the creation and reuse of `Mapper` instances
+ * for different types. Each type of `Mapper` is stored in a map, ensuring that only one instance
+ * per type exists during the application's lifecycle.
+ */
+export class MapperProvider {
+  /**
+   * A static map that stores singleton instances of `Mapper` objects, keyed by their type name.
+   */
+  // eslint-disable-next-line @typescript-eslint/consistent-generic-constructors, @typescript-eslint/no-explicit-any
+  private static readonly MAPPER_INSTANCES: Map<string, Mapper<any>> = new Map();
```

Comment:
OK, replace this any with unknown as well

Reply:
Done